baremetal/metal3datatemplate_manager.go

```diff
@@ -341,40 +343,87 @@ func (m *DataTemplateManager) createData(ctx context.Context,
 	return indexes, nil
 }
 
+// Handles errors generated by Metal3Data retrieval.
+// First return value is true of Metal3Data was retrieved successfully.
+// Second return value contains an error message if an unknown error was encountered.
+func (m *DataTemplateManager) handleDeleteDataError(err error, filter string, dataClaimName string, dataName string) (bool, string) {
+	if err != nil && apierrors.IsNotFound(err) {
+		m.Log.Error(err, "Transient error", "Metal3Data was not found based on filter:", filter, "Metal3DataClaim", dataClaimName, "Metal3DataTemplate", m.DataTemplate.Name, "Metal3Data", dataName)
```

Comment:
Why is it transient error? Is this for cases when it was already deleted, so we are just checking to make sure?